src/main.cpp

```diff
@@ -4850,61 +4862,68 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                 // Pruned nodes may have deleted the block, so check whether
                 // it's available before trying to send.
                 if (send && (mi->second->nStatus & BLOCK_HAVE_DATA))
-                {
+                  {
                     // Send block from disk
                     CBlock block;
                     if (!ReadBlockFromDisk(block, (*mi).second, consensusParams))
```

Comment:
You need to check for mi->second->nStatus & BLOCK_HAVE_DATA before running this.